xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -594,89 +595,144 @@ void CGUIDialogAddonInfo::OnSettings()
 
-    if (info_addon)
+    // Find add-on in local installation
+    if (!CServiceBroker::GetAddonMgr().GetAddon(dep.id, addonInstalled))
     {
-      CFileItemPtr item(new CFileItem(info_addon->Name()));
-      std::stringstream str;
-      str << it.id << " " << it.versionMin.asString() << " -> " << it.version.asString();
-      if ((it.optional && !local_addon) || (!it.optional && local_addon))
-        str << " "
-            << StringUtils::Format(g_localizeStrings.Get(39022).c_str(),
-                                   local_addon ? g_localizeStrings.Get(39019).c_str()
-                                               : g_localizeStrings.Get(39018).c_str());
-      else if (it.optional && local_addon)
-        str << " "
-            << StringUtils::Format(g_localizeStrings.Get(39023).c_str(),
-                                   g_localizeStrings.Get(39019).c_str(),
-                                   g_localizeStrings.Get(39018).c_str());
-
-      item->SetLabel2(str.str());
-      item->SetArt("icon", info_addon->Icon());
-      item->SetProperty("addon_id", it.id);
-      items.Add(item);
+      addonInstalled = nullptr;
+      allDepsInstalled = false;
     }
-    else
+
+    // Find add-on in repositories
+    if (!CServiceBroker::GetAddonMgr().FindInstallableById(dep.id, addonAvailable))
     {
-      CFileItemPtr item(new CFileItem(it.id));
-      item->SetLabel2(g_localizeStrings.Get(10005)); // Not available
-      items.Add(item);
+      addonAvailable = nullptr;
     }
+
+    installedWithAvailable[dep.id] = {addonInstalled, addonAvailable};
   }
 
-  CFileItemPtr backup_item = GetCurrentListItem();
-  while (true)
+  if (!allDepsInstalled || showIfAllDepsInstalled == ShowIfAllDepsInstalled::YES)
   {
-    pDialog->Reset();
-    pDialog->SetHeading(reactivate ? 39024 : 39020);
-    pDialog->SetUseDetails(true);
-    for (auto& it : items)
-      pDialog->Add(*it);
-    pDialog->EnableButton(!reactivate, 186);
-    pDialog->SetButtonFocus(true);
-    pDialog->Open();
-
-    if (pDialog->IsButtonPressed())
-      return true;
+    auto pDialog = CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+        WINDOW_DIALOG_SELECT);
+    CFileItemList items;
+
+    for (auto it : deps)
+    {
+      // All combinations of dep_addon and local_addon validity are possible and information
+      // must be displayed even when there is no dep_addon.
+      // info_addon is the add-on to take the information to display (name, icon) from. The
+      // version in the repository is preferred because it might contain more recent data.
+
+      auto instWithAvail = installedWithAvailable.find(it.id);
+      std::shared_ptr<IAddon> info_addon = instWithAvail->second.m_available
+                                               ? instWithAvail->second.m_available
+                                               : instWithAvail->second.m_installed;
+
+      if (CAddonType::IsVisibleInDialog(info_addon))
```

Comment:
see my comment above. AddonType is a very basic core type. It should not have any GUI related methods. Maybe just bad wording?

Reply:
`if (infoAddon->MainType() != ADDON_SCRIPT_MODULE`

Reply:
yes, sounds all reasonable, then i'll go for a rework :) np

Reply:
yes, please

Reply:
and please a comment explaining why script modules (whatever this is) are excluded.